### DISCUSSION TITLE
Resource tagging

### DISCUSSION THREAD

**Discussion by thodson-usgs** [nebari-dev/nebari#2212] [Q&A]
Hello, 
My agency (USGS) is trying to set up a Nebari deployment, but we're required to tag all resources on our AWS account. 
We've found some clues about how to do this, like [accounting/variables.tf](template/aws/modules/accounting/variables.tf) and [overrides](https://www.nebari.dev/docs/explanations/custom-overrides-configuration/#terraform). However, overrides seem extreme for such a basic operation. Is there a better way to pass a set of resource tags?

**Comment by thodson-usgs** [comment]
After further digging, I found [discussion 2097](https://github.com/orgs/nebari-dev/discussions/2097). Reading that makes me a bit more comfortable with the overrides option, but guidance about best practice would still be welcome.

**Comment by ronald50928** [comment]
As mentioned above. Guidance is minimal in current AWS deployment guides. We (USGS) propose the following to be added to the deployment guides. This can be added under the [Permissions Boundary (Optional)](https://www.nebari.dev/docs/explanations/provider-configuration#permissions-boundary-optional) session.

_Tagging (Optional)
AWS allows you to assign metadata to many of your AWS resources in the form of tags. Each tag is a simple label consisting of a key and an optional value to store information about the resource or data retained on that resource. . Implementing a consistent tagging strategy can make it easier to filter and search for resources, monitor cost and usage, and manage your Nebari environment running on AWS.

Nebari supports setting tags while deploying Nebari to be applied on all the resources created by Nebari. Here is an example of how you would set permissions boundary in nebari-config.yaml._

`amazon_web_services:
#tags can be added by using <key>: "<value>"
      tags:
          Project: "Science"
          Owner: "john.doe"
          CostCode: "12345"`
          
@aktech

**Reply by dharhas** [reply]
cc: @pavithraes - docs
